libraries/AP_HAL_PX4/I2CDevice.cpp

```diff
@@ -17,16 +17,54 @@
 #include <AP_HAL/AP_HAL.h>
 
 #include "Util.h"
+#include "Scheduler.h"
+
+extern bool _px4_thread_should_exit;
 
 namespace PX4 {
 
 uint8_t PX4::PX4_I2C::instance;
 
 /*
+  constructor for I2C wrapper class
+ */    
+PX4_I2C::PX4_I2C(uint8_t bus) :
+  I2C(devname, devpath, map_bus_number(bus), 0, 400000UL)
+{}
+
+/*
+  map ArduPilot bus numbers to PX4 bus numbers
+ */    
+uint8_t PX4_I2C::map_bus_number(uint8_t bus) const
+{
+    switch (bus) {
+    case 0:
+        // map to internal bus
+#ifdef PX4_I2C_BUS_ONBOARD
```

Comment:
Also, if it's not defined, shouldn't you return an invalid number to mean "bus not available"?  Looking at the source code PX4-V4 doesn't have an onboard bus, only an external one... AFAICS we would return a wrong number here.